app/views/paths/index.html.erb

```diff
@@ -16,11 +16,18 @@
                 <h2 class="text-3xl font-medium text-gray-800 dark:text-gray-300">Foundations</h2>
               <% end %>
             </div>
+            <% if current_user.present? && current_user.on_path?(@foundations) %>
+              <div class="absolute top-3.5 right-3.5 !mt-0 md:static">
+                <%= render Ui::BadgeComponent.new(color: 'green') do %>
+                  <%= @badge_text %>
+                <% end %>
+              </div>
+            <% end %>
           </div>
 
           <div class="space-x-6 justify-center hidden md:flex">
             <%= render Paths::SelectButtonComponent.new(current_user:, path: @foundations, classes: 'px-12') %>
-            <%= render Paths::ViewButtonComponent.new(current_user:, path: @foundations, classes: 'px-12') %>
+            <%= render Paths::ViewButtonComponent.new(current_user:, path: @foundations, classes: "px-12 #{'hidden' if current_user.present? && current_user.on_path?(@foundations)}") %>
```

Comment:
Rather than this `"hidden"` class interpolation, I think it'd be better to use ViewComponent's `#render?`. Dries things up and gives this control to the component itself.

So you'd have something like:
```rb
module Paths
  class ViewButtonComponent < ApplicationComponent
    # ... the code that's already here
    
    def render?
      current_user.blank? || !current_user.on_path?(path)
    end
  end
end
```
ViewComponent will just not render the component at all under those conditions. Here's the [documentation for this](https://viewcomponent.org/guide/conditional_rendering.html). Let me know if you have any questions.

Reply:
Thanks @JoshDevHub I'll look into these.

Reply:
Hi @JoshDevHub I have made the requested changes. Thanks for taking the time to review these changes.